src/openassetio-python/package/openassetio/managerApi/ManagerInterface.py

```diff
@@ -479,85 +479,104 @@ def finalizedEntityVersion(self, entityRefs, context, hostSession, overrideVersi
-        same length, and it should be assumed that it is a 1:1 mapping
-        of a relationship definition to an entity. If this is not the
-        case, ValueErrors should be thrown.
+        references should not be considered meaningful, but the outer
+        list should match the input entity reference order.
 
         If any relationship definition is unknown, then an empty list
-        should be returned for that relationship, and no errors should
-        be raised.
-
-        @param entityRefs List[` @fqref{EntityReference}
-        "EntityReference" `]
+        should be returned for that entity, and no errors raised. The
+        default implementation returns an empty list for all
+        relationships.
 
-        @param relationshipTraitsDatas `List[`
-        @fqref{TraitsData} "TraitsData" `]`
+        @param relationshipTraitsDatas `List[` @fqref{TraitsData} "TraitsData" `]`
+        The traits of the relationships to query.
 
-        @param context Context The calling context.
+        @param toEntityReference @fqref{EntityReference} "EntityReference"
+        The @ref entity_reference to query the specified relationships
+        for.
 
         @param hostSession openassetio.managerApi.HostSession The host
         session that maps to the caller, this should be used for all
         logging and provides access to the openassetio.managerApi.Host
         object representing the process that initiated the API session.
 
-        @param resultTraitSet `Set[str]` or None, a hint as to what
-        traits the caller is expecting the returned entities to have.
-
-        @return List[List[str]] This MUST be the correct length,
-        returning an empty outer list is NOT valid. (ie: max(len(refs),
-        len(relationships)))
-
-        @exception ValueError If more than one reference and
-        relationship are provided, but they lists are not equal in
-        length, ie: not a 1:1 mapping of entities to relationships. The
-        abstraction of this interface into the Manager class does
-        cursory validation that this is the case before calling this
-        function.
+        @param context Context The calling context.
 
-        @unstable
+        @param resultTraitSet `Set[str]` or None, a hint as to what
+        traits the returned entities should have.
+
+        @return `List[List[@fqref{EntityReference} "EntityReference"`]]`
+        A list of references to related entities, for each input
+        relationship.
+
+        @note Ensure that your implementation of
+        @fqref{managerApi.ManagerInterface.managementPolicy}
+        "managementPolicy" responds appropriately when queried with
+        relationship trait sets and a read policy to communicate to the
+        host whether or not you are capable of handling queries for
+        those relationships in this method.
         """
-        raise NotImplementedError
+        return [[] for _ in relationshipTraitsDatas]
```

Comment:
[Question] Just checking that python is the appropriate place for these default implementations, rather than C++? I'm not totally sure what the pattern is currently, and I know you gestured at doing python first and getting that working before moving down, so perhaps this is an in-progress stopgap?

Reply:
Yeah - we haven't migrated to C++ yet (#913), after the last round, we wanted to try and have the python forms up to date first, rather than doing too many things at once in the C++ port.

Reply:
:+1: